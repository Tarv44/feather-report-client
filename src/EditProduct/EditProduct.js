import React, {Component} from 'react';
import styles from './EditProduct.module.css';
import EditFeatures from '../EditFeatures/EditFeatures';

export default class EditProduct extends Component {
    static defaultProps = {
        product: {
            title: '',
            price: 0,
            description: '',
            category: 'Select Category',
            features: [],
            link: '',
            id:''
        }
    }

    state = {
        title: this.props.product.title,
        price: this.props.product.price,
        description: this.props.product.description,
        category: this.props.product.category,
        features: this.props.product.features,
        link: this.props.product.link,
        id: this.props.product.id
    }

    updateTitle(title) {
        this.setState({ title })
    }

    updatePrice(price) {
        this.setState({ price })
    }

    updateDescription(description) {
        this.setState({ description })
    }

    updateLink(link) {
        this.setState({ link })
    }

    updateCategory(category) {
        this.setState({ category })
    }
    
    removeFeature = (e, i) => {
        e.preventDefault()
        const features = this.state.features
        features.splice(i, 1)
        this.setState({ features })
    }

    addFeature = (e) => {
        e.preventDefault()
        const features = this.state.features
        features.push({message: '', id: ''})
        this.setState({ features })
    }

    render() {
        const id = this.state.id

        return (
            <form className={styles.productForm}>
                <h2>Edit Product</h2>
                <fieldset>
                    <legend>Details</legend>
                    <label htmlFor={`productTitle-${id}`}>Title</label>
                    <input 
                        type="text" 
                        name="title" 
                        id={`productTitle-${id}`}
                        value={this.state.title}
                        onChange={e => this.updateTitle(e.target.value)} 
                    />
                    <label htmlFor={`productPrice-${id}`}>Price</label>
                    <input 
                        type="text" 
                        name="price" 
                        id={`productPrice-${id}`}
                        value={this.state.price}
                        onChange={e => this.updatePrice(e.target.value)} 
                    />
                    <label htmlFor={`productDescr-${id}`}>Description</label>
                    <textarea
                        name="description" 
                        id={`productDescr-${id}`}
                        value={this.state.description}
                        rows={5}
                        onChange={e => this.updateDescription(e.target.value)} 
                    />
                    <label htmlFor={`productLink-${id}`}>Buy Online Link</label>
                    <input
                        name="link" 
                        id={`productLink-${id}`}
                        value={this.state.link}
                        onChange={e => this.updateLink(e.target.value)} 
                    />
                </fieldset>
                <EditFeatures 
                    category={this.state.category}
                    features={this.state.features}
                    id={id}
                    handleCategory={this.updateCategory}
                    removeFeature={this.removeFeature}
                    addFeature={this.addFeature}
                />
            </form>
        )
    }
}