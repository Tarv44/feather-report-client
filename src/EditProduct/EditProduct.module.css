.form {
    display: flex;
    flex-direction: column;
    align-items: center;
    border: solid black 3px;
    min-width: 100%;
    padding-bottom: 20px;
    margin-top: 10px;
}



.form fieldset {
    width: 90%;
}

.submit {
    font-size: 18px;
    margin-top: 1em;
    background-color: #26547C;
    color: #fff;
    border: solid black 2px;
    border-radius: 3px;
    padding: .3em 1em;
    box-shadow: 0 1px 10px 3px rgba(0,0,0,.1);
}