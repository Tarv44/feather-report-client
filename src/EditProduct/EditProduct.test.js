import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import EditProduct from './EditProduct'
import { categories } from '../component.fixtures'

describe('----- EditProduct -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <EditProduct
                allCats={categories}
            />
        , div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <EditProduct
                    allCats={categories}
                />
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})