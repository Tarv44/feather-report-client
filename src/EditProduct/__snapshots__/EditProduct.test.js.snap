// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- EditProduct ----- renders the UI as expected 1`] = `
<form
  className="form"
>
  <h2>
    Add Product
  </h2>
  <fieldset>
    <legend>
      Details
    </legend>
    <label
      htmlFor="productTitle-null"
    >
      Title
    </label>
    <input
      id="productTitle-null"
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="productPrice-null"
    >
      Price
    </label>
    <input
      id="productPrice-null"
      name="price"
      onChange={[Function]}
      type="text"
      value={0}
    />
    <label
      htmlFor="productDescr-null"
    >
      Description
    </label>
    <textarea
      id="productDescr-null"
      name="description"
      onChange={[Function]}
      rows={5}
      value=""
    />
    <label
      htmlFor="productLink-null"
    >
      Buy Online Link
    </label>
    <input
      id="productLink-null"
      name="link"
      onChange={[Function]}
      type="text"
      value=""
    />
  </fieldset>
  <fieldset>
    <legend>
      Category and Features
    </legend>
    <h3
      className="label"
    >
      Category
    </h3>
    <div
      className="category"
    >
      <select
        onChange={[Function]}
        value="Select Category"
      >
        <option>
          Select Category
        </option>
        <option
          id={1}
        >
          Knives
        </option>
        <option
          id={2}
        >
          Blenders
        </option>
        <option
          id={3}
        >
          Toasters
        </option>
      </select>
      <button
        className="button"
        onClick={[Function]}
      >
        Create New Category
      </button>
    </div>
    <h3
      className="label"
    >
      Features
    </h3>
    <button
      className="addFeat"
      onClick={[Function]}
    >
      Add Another Feature
    </button>
  </fieldset>
  <button
    className="submit"
    onClick={[Function]}
  >
    Add Product
  </button>
</form>
`;
